common/path/platform/c++17/path.cpp

```diff
@@ -42,14 +42,25 @@ bool IsAbsolute(const string &path) {
 	return fs::path(path).is_absolute();
 }
 
-expected::ExpectedUnorderedSet<string> ListFiles(
+expected::ExpectedBool IsExecutable(const string &file_path) {
+	fs::perms perms = fs::status(file_path).permissions();
```

Comment:
It is/should be cheaper to do this when iterating over the directory contents. That's why I avoided this in the inventory parser code.

Reply:
Hmm, it is done when iterating over the contents. It is passed in as a function predicate to the `ListFiles` (which maybe should be `CollectFiles` or something similar?

Reply:
That's not exactly what I meant -- when you iterate over the directory, you already have the names together with the `stat` structs for the entries. So you can check the permissions without any extra `stat()` syscall.

Reply:
I see, but this way we can iterate just once for two conditions though.

I'm keeping this one since it's flexible. We can optimise later, if this ever becomes a problem